### DISCUSSION TITLE
Convert json list to C++ list

### DISCUSSION THREAD

**Discussion by RPowell-C** [nlohmann/json#4329] [Q&A] [answered]
I have a json file that looks like: 
```
{
    "affected_files": ["./phoibe.py", "json-files/ucal.json"],
    "affects_settings": true
}

```
my main function looks like:
```
int main()
{
    std::ifstream file("update.json");
    json data = json::parse(file);
    // setting this to a string for nowm, will change to array later
    json affected_files = json::array();
    // going ahead and creating the array
    // 105 entries is the current lines in the settings.json file
    // create vector to hopefully store the value of the json array as a C++ array




    affected_files = data["affected_files"];
    std::cout << affected_files << std::endl;

    // download();

    std::cout << "Done!" << std::endl;

}
```
I may just be stupid, but I can get `affected_files` to print, but it's a literal, and I can't apply `_json` because it'll read it as `affected_files_json`, I am brand new to C++ so there may be a way to do that that I don't know of, but I want to be able to read the contents of the json list as a C++ list

**Comment by RPowell-C** [comment, marked as answer]
I solved my own problem lol, I initialized a vector, and then read through the parsed (i think) json list and added it to the vector, here is the code for future people who may have this problem 
```
std::vector<std::string> affected_files_array;
    int counter = 0;
    for (std::string x : data["affected_files"])
    {
        affected_files_array.push_back(x);
        counter++;

    }


```

**Comment by gregmarr** [comment]
Here is a much better way:

    auto data = nlohmann::json::parse(R"({
      "affected_files": ["./phoibe.py", "json-files/ucal.json"],
      "affects_settings": true
    })");
    auto affected_files = data["affected_files"].get<std::vector<std::string>>();

    for (auto &s : affected_files)
    {
        std::cout << s << "\n";
    }

**Reply by RPowell-C** [reply]
Thanks, I'll try it out
